app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,47 @@
+module SessionsHelper
+    # Logs in the given user.
+    def log_in(user)
+        session[:user_id] = user.id
+    end
+
+    # Remembers a user in a persistent session.
+    def remember(user)
+        user.remember
+        cookies.permanent.signed[:user_id] = user.id
+        cookies.permanent[:remember_token] = user.remember_token
+    end
+
+    # Returns the current logged-in user (if any).
+    def current_user
+        if (user_id = session[:user_id])
```

Comment:
Do you understand these lines: `if (user_id = session[:user_id])` and `elsif (user_id = cookies.signed[:user_id])` ?

I ask because I think they are confusing (and I know they're straight out of Hartl's tutorial). To me, they read as though they are simple conditions when in fact they are checking if the second value exists and assigning it to `user_id` - not very easy to read. I'd recommend rewriting this to make it easier to follow, even if it requires a little more repetition. 

In a highly performance-tuned application, something like this might be needed, but that's not the main concern here.